internal/controller/warehouses/git.go

```diff
@@ -139,14 +146,47 @@ func (r *reconciler) selectTagAndCommitID(
 ) (string, string, error) {
 	if sub.CommitSelectionStrategy == kargoapi.CommitSelectionStrategyNewestFromBranch {
 		// In this case, there is nothing to do except return the commit ID at the
-		// head of the branch.
+		// head of the branch unless there are scanPaths/ignorePaths configured to
+		// handle.
 		commit, err := r.getLastCommitIDFn(repo)
-		return "", commit, errors.Wrapf(
-			err,
-			"error determining commit ID at head of branch %q in git repo %q",
-			sub.Branch,
-			sub.RepoURL,
-		)
+		if err != nil {
+			return "", "",
+				errors.Wrapf(err, "error determining commit ID at head of branch %q in git repo %q",
+					sub.Branch,
+					sub.RepoURL)
+		}
+		// below if statement handles scanPaths/ignorePaths filters if configured in a
+		// git subscription
+		if sub.ScanPaths != nil || sub.IgnorePaths != nil {
+
+			// getting actual diffPaths for last commit
+			diffs, err := r.getDiffPathsForCommitIDFn(repo, commit)
```

Comment:
I understood the intent of this function call to be finding all paths that have changed since `commit`, but the value of `commit` here is that of the HEAD. Wouldn't a comparison of HEAD to HEAD always yield no diffs.

Reply:
My approach was wrong, now I'm getting it - we need to obtain diffs between the commit that produced last Freight and the HEAD as you say, so Depth of clone loses its sense probably as the mentioned Freight-producing commit is at unknown Depth... Thanks a lot for your review @krancour ! I will reiterate.